lib/internal/modules/esm/loader.js

```diff
@@ -507,20 +507,23 @@ class ESMLoader {
    */
   async load(url, context = {}) {
     const loaders = this.#loaders;
-    let i = loaders.length - 1;
+    let hookIndex = loaders.length - 1;
     let {
       fn: loader,
       url: loaderFilePath,
-    } = loaders[i];
-    let chainFinished = i === 0;
+    } = loaders[hookIndex];
+    let chainFinished = hookIndex === 0;
+
+    const next = (nextUrl, ctx = context) => {
+      // `next` has been called, so decrement our pointer
+      --hookIndex;
```

Comment:
actually, maybe yes this is better separated for someone who doesn't know the subtle difference between `--hookIndex` and `hookIndex--`

Reply:
And when it’s on its own line, the different doesn’t matter.

Reply:
true. too late—you missed the ship 😜